### ISSUE TITLE
Allow internal modules to be tested

### ISSUE THREAD

**Issue by martijnbastiaan** [martijnbastiaan/doctest-parallel#87]
As mentioned in the migration guide, `doctest-parallel` can only test exported binders:

-------------

![Screenshot 2024-12-14 at 11-02-12 Migration guide for users coming from `doctest` · Issue #11 · martijnbastiaan_doctest-parallel](https://github.com/user-attachments/assets/7b2f1fb7-ed81-4ca8-96c7-d2010bbfe3f4)

-------------

It would be nice if modules could be marked to run in "interpreted mode", i.e.:

```
  {-# ANN module "doctest-parallel: --interpret" #-}
```

By copying these internal modules to a temporary directory and using GHCi to interpret them, we could get the best of both worlds: run as many examples as possible against compiled/optimized code, while also being able to run doctests on internal ones. (We don't necessarily have to copy them, but I'm unsure how thread-safe starting multiple GHCi sessions is..)

**Comment by tmplt** [comment]
Do I understand correctly that a feature like this is required to test non-exported functions, too? E.g.

```haskell
module SomeHidden (funA) where

funA = funB

-- | Test non-exported function
-- >>> funB
-- True
funB = True
```

**Comment by martijnbastiaan** [comment]
Yes, that's correct. In your example, only `funA` is accessible when loading compiled code.
